File: Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace aksjehandel.Models
{
    public class Order
    {
        public int Id { get; set; }
        public int CompanyId { get; set; }
        public string CompanyName { get; set; }
        public string CompanySymbol { get; set; }
        public int PortfolioId { get; set; }
        public string PortfolioDisplayName { get; set; }
        public double PortfolioCash { get; set; }
        [RegularExpression(@"^(buy|sell)$")]
        public string Type { get; set; }
        [RegularExpression(@"[0-9]{2,8}")]
        public double Price { get; set; }
        [RegularExpression(@"[0-9]{1,8}")]
        public int Amount { get; set; }
    }
}

File: aksjehandel-react/client-app/src/types/order.ts
export interface ServerOrder {
    id: number;
    companyId: number;
    companyName: string;
    companySymbol: string;
    portfolioId: number;
    portfolioDisplayName: string;
    portfolioCash: number;
    type: "buy" | "sell";
    price: number;
    amount: number;
}
export interface NewOrder {
    companyId: number;
    portfolioId: number;
    type: "buy" | "sell";
    price: number;
    amount: number;
}
export interface EditOrder {
    id: number;
    price: number;
    amount: number;
}